cmd/antrea-agent/options.go

```diff
@@ -104,85 +112,24 @@ func (o *Options) validate(args []string) error {
-		// (iptables/ipvs) in the root netns. If the Endpoint is not local the DNATed traffic will be output to
-		// the physical network directly without going back to OVS for Egress NetworkPolicy enforcement, which
-		// breaks basic security functionality. Therefore, we usually do not allow the NoEncap traffic mode without
-		// AntreaProxy. But one can bypass this check and force this feature combination to be allowed, by defining
-		// the ALLOW_NO_ENCAP_WITHOUT_ANTREA_PROXY environment variable and setting it to true. This may lead to
-		// better performance when using NoEncap if Egress NetworkPolicy enforcement is not required.
-		if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
-			if env.GetAllowNoEncapWithoutAntreaProxy() {
-				klog.InfoS("Disabling AntreaProxy in NoEncap mode will prevent Egress NetworkPolicy rules from being enforced correctly")
-			} else {
-				return fmt.Errorf("TrafficEncapMode %s requires AntreaProxy to be enabled", o.config.TrafficEncapMode)
-			}
-		}
-		if encryptionMode != config.TrafficEncryptionModeNone {
-			return fmt.Errorf("TrafficEncryptionMode %s may only be enabled in %s mode", encryptionMode, config.TrafficEncapModeEncap)
-		}
-	}
-	if o.config.NoSNAT && !(encapMode == config.TrafficEncapModeNoEncap || encapMode == config.TrafficEncapModeNetworkPolicyOnly) {
-		return fmt.Errorf("noSNAT is only applicable to the %s mode", config.TrafficEncapModeNoEncap)
-	}
-	if encapMode == config.TrafficEncapModeNetworkPolicyOnly {
-		// In the NetworkPolicyOnly mode, Antrea will not perform SNAT
-		// (but SNAT can be done by the primary CNI).
-		o.config.NoSNAT = true
-	}
-	if err := o.validateAntreaProxyConfig(); err != nil {
-		return fmt.Errorf("proxy config is invalid: %w", err)
-	}
-	if err := o.validateFlowExporterConfig(); err != nil {
-		return fmt.Errorf("failed to validate flow exporter config: %v", err)
-	}
-	if features.DefaultFeatureGate.Enabled(features.Egress) {
-		for _, cidr := range o.config.Egress.ExceptCIDRs {
-			_, _, err := net.ParseCIDR(cidr)
-			if err != nil {
-				return fmt.Errorf("Egress Except CIDR %s is invalid", cidr)
-			}
-		}
-	}
-	if features.DefaultFeatureGate.Enabled(features.NodePortLocal) {
-		startPort, endPort, err := parsePortRange(o.config.NodePortLocal.PortRange)
-		if err != nil {
-			return fmt.Errorf("NodePortLocal portRange is not valid: %v", err)
-		}
-		o.nplStartPort = startPort
-		o.nplEndPort = endPort
-	} else if o.config.NodePortLocal.Enable {
-		klog.InfoS("The nodePortLocal.enable config option is set to true, but it will be ignored because the NodePortLocal feature gate is disabled")
-	}
-	if err := o.validateAntreaIPAMConfig(); err != nil {
-		return fmt.Errorf("failed to validate AntreaIPAM config: %v", err)
+	if strings.EqualFold(config.ExternalNode.String(), o.config.NodeType) {
+		o.nodeType = config.ExternalNode
+		return o.validateVMOptions()
+	} else if strings.EqualFold(o.config.NodeType, config.K8sNode.String()) {
```

Comment:
nit: use same order as above check, and do we really need to convert the type from string to int and support both uppercase and lowercase?

Reply:
@jianjuns suggested using enum (int)  for comparation should be more efficient. Actually, we only compare string value with configuration here, and we can use int type to compare in the functions.